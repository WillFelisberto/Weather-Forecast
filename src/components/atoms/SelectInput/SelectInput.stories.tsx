import type { Meta, StoryObj } from '@storybook/react';

import { SelectInput } from './SelectInput';

const meta: Meta<typeof SelectInput> = {
  title: 'Atoms/SelectInput',
  component: SelectInput,
  tags: ['autodocs'],
  args: {
    options: ['Option 1', 'Option 2', 'Option 3']
  }
};

export default meta;

type Story = StoryObj<typeof SelectInput>;

export const Default: Story = {};

export const WithCustomClass: Story = {
  args: {
    className: 'border border-blue-500'
  }
};
