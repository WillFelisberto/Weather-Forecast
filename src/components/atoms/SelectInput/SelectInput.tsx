import clsx from 'clsx';
import { ChevronDown } from 'lucide-react';
import { useState } from 'react';

interface SelectInputProps {
  options: string[];
  className?: string;
}

export const SelectInput: React.FC<SelectInputProps> = ({ options, className }) => {
  const [selectedOption, setSelectedOption] = useState(options[0]);
  const [isOpen, setIsOpen] = useState(false);

  return (
    <div className={clsx('flex items-center space-x-4', className)}>
      <div className="relative">
        <button
          className="flex items-center justify-between w-48 bg-gray-800 text-white px-4 py-2 rounded-lg border border-gray-700 hover:bg-gray-700 transition"
          onClick={() => setIsOpen((prev) => !prev)}
        >
          <span>{selectedOption}</span>
          <ChevronDown size={16} />
        </button>
        {isOpen && (
          <div className="absolute mt-2 w-full bg-gray-800 rounded-lg shadow-lg z-10">
            {options.map((option, idx) => (
              <button
                key={idx}
                onClick={() => {
                  setSelectedOption(option);
                  setIsOpen(false);
                }}
                className="block px-4 py-2 text-white hover:bg-gray-700 text-left w-full"
              >
                {option}
              </button>
            ))}
          </div>
        )}
      </div>
    </div>
  );
};
