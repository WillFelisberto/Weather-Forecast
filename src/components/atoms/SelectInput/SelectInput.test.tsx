// src/components/atoms/SelectInput/SelectInput.test.tsx
import { fireEvent, render, screen } from '@/tests/test-utils';

import { SelectInput } from './SelectInput';

describe('SelectInput', () => {
  const options = ['Option 1', 'Option 2', 'Option 3'];

  it('renders the select input with the default selected option', () => {
    const { container } = render(<SelectInput options={options} />);

    // Verifica se a primeira opção está selecionada por padrão
    expect(screen.getByText('Option 1')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('opens the dropdown menu when the button is clicked', () => {
    render(<SelectInput options={options} />);

    // O dropdown deve estar inicialmente fechado
    expect(screen.queryByText('Option 2')).not.toBeInTheDocument();

    // Clica no botão para abrir o dropdown
    fireEvent.click(screen.getByText('Option 1'));

    // Verifica se as opções são exibidas
    expect(screen.getByText('Option 2')).toBeInTheDocument();
    expect(screen.getByText('Option 3')).toBeInTheDocument();
  });

  it('selects a new option when it is clicked', () => {
    render(<SelectInput options={options} />);

    // Abre o dropdown
    fireEvent.click(screen.getByText('Option 1'));

    // Clica na segunda opção
    fireEvent.click(screen.getByText('Option 2'));

    // Verifica se a opção foi selecionada
    expect(screen.getByText('Option 2')).toBeInTheDocument();
    expect(screen.queryByText('Option 1')).not.toBeInTheDocument();
  });

  it('closes the dropdown menu after selecting an option', () => {
    render(<SelectInput options={options} />);

    // Abre o dropdown
    fireEvent.click(screen.getByText('Option 1'));

    // Clica na segunda opção
    fireEvent.click(screen.getByText('Option 2'));

    // Verifica se o dropdown foi fechado
    expect(screen.queryByText('Option 3')).not.toBeInTheDocument();
  });
});
