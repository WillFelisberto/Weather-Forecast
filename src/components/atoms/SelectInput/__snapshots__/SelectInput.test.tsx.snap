// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectInput renders the select input with the default selected option 1`] = `
<div>
  <div
    class="flex items-center space-x-4"
  >
    <div
      class="relative"
    >
      <button
        class="flex items-center justify-between w-48 bg-gray-800 text-white px-4 py-2 rounded-lg border border-gray-700 hover:bg-gray-700 transition"
      >
        <span>
          Option 1
        </span>
        <svg
          class="lucide lucide-chevron-down"
          fill="none"
          height="16"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m6 9 6 6 6-6"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
